frontend/src/components/yugioh/table/market/YugiohCardMarket.tsx

```diff
@@ -60,4 +64,49 @@ function YugiohCardMarket(props: YugiohCardMarketProps): JSX.Element {
   );
 }
 
+function TableWithNoData(): JSX.Element {
+  return (
+    <div className="text-center">
+      <div className="hidden lg:flex flex-col md:items-center justify-center overflow-auto">
+        <MarketTable className="w-11/12 md:w-full lg:w-5/6 mb-12">
+          <thead className="text-sm lg:text-base">
+            <tr>
+              <th colSpan={3}>Seller</th>
+              <th colSpan={2}>Card details</th>
+              <th>Available</th>
+              <th colSpan={3}>Buy</th>
+            </tr>
+          </thead>
+          <tfoot>
+            <tr>
+              <td colSpan={9}>
+                There are currently no listings for this card. You can be the first one!
+              </td>
+            </tr>
+          </tfoot>
+        </MarketTable>
+      </div>
+      <div className="flex lg:hidden flex-col mx-auto overflow-auto">
```

Comment:
Isn't this whole div with its content a copy of the one above? Does it do anything? I also see this for the return of `YugiohCardMarket`.

Reply:
the duplicates are there because some of the cells have a different ``colSpan`` depending on the viewport (which cannot be controlled with CSS classes alone). This necessity comes from the fact that some cells are hidden on smaller viewports and this would affect the table layout significantly, so this solution came into existence.

Reply:
As for this specific case, there's also the fact that the pagination is controlled with props and the pagination is placed in a ``tfoot``, so there would be no way to insert another cell next to the pagination (+ there's no point in having the widget if there's nothing to paginate)